packages/rainbowkit/src/components/ConnectButton/ConnectButtonRenderer.tsx

```diff
@@ -133,7 +136,7 @@ export function ConnectButtonRenderer({
               iconUrl: resolvedChainIconUrl,
               id: chainId,
               name: chainName,
-              unsupported: !isCurrentChainSupported,
+              unsupported: isConnected && !isCurrentChainSupported,
```

Comment:
We only show `unsupported` if a user is connected and if the chain is not supported